packages/dds/tree/docs/wip/cell-model.md

```diff
@@ -0,0 +1,210 @@
+This forwarding information is used to represent move information: the source cell is annotated with forwarding information that points to the destination cell.
+
+In addition to their storage role, cells act as markers relative to which an edit may be performed (e.g., "Before/after cell foo").
+
+### Operations
+
+The model admits the following operations:
+
+> WIP: An alternative set of operations based on hiding cells instead of clearing them is being considered.
+> The general structure of the model is unaffected.
+
+- <u>Allocate</u> a new cell at a specific location in the sequence
+
+- <u>Fill</u> a cell with content (overwriting existing content if any)
+
+- <u>Clear</u> content from a cell, leaving it empty
+
+- <u>Add a forwarding annotation</u> to a cell
+
+- <u>Remove a forwarding annotation</u> from a cell
+
+Note that cells cannot be deallocated (or moved).
+This reflects the fact that a client may perform an edit relative to some content's location at a point in time, even if that content has since been deleted or moved: the client is performing the edit relative to the cell which contained that content.
+
+## Building on the Model
+
+The model can be used to describe low-level edits that a client may perform on a field in a collaborative environment and by so doing imply some of their merge semantics.
+
+We provide here two lists of such low-level edits, one for fields whose number of cells is fixed, and one for fields whose number of cells is dynamic.
+
+Note that the fixed and dynamic characteristics here apply only to the number of cells, not to the number of elements.
+For example, an optional field, which can contain zero or one element, is a fixed-sized field composed of exactly one cell.
+That cell exists even when the field is not populated with an element. 
+
+The dichotomy has no basis in the cell model but reflects what we think is a sensible separation and matches industry-standard patterns of editing.
+
+### Edits on Dynamically-Sized Fields
+
+Dynamically-sized fields can be thought of as having list-like behavior: elements can be added anywhere in the list and removed at will which makes the list contents dynamically grow and shrink (while the number of cells in the list only grows).
+Dynamically-sized fields typically start out empty of any cell.
+
+The low-level edits that such fields might support can be decomposed as follows:
+
+- Insert: allocate and fill a cell
+
+- Delete/Remove: clear a cell
+
+- Revive: fill a (cleared) cell with the contents it contained before
+
+- Replace: fill a (filled) cell
+
+- Move content from A to B:
+  
+  - A: clear a (filled) cell and add a forwarding annotation to it
+  
+  - B: allocate and fill a cell with the content being moved
+
+- Return content to A from B:
+  
+  - A: remove the forwarding annotation from a cell and fill it with the returned content
+  
+  - B: clear the (filled) cell
+
+### Edits on Fixed-Sized Fields
```

Comment:
Are the move semantics afforded by dynamically sized fields not useful for fixed-sized arrays? It seems like (unless I'm missing something), a consumer who opted to use a dynamically sized list for some semantically fixed-sized contents would get different merge behaviors than if they used the fixed-sized array.

Reply:
Move semantics are useful for fixed sized fields as well. I updated the document to reflect that.

You're right that a consumer who opted to use a dynamically sized list for some semantically fixed-sized contents would get different merge behaviors than if they used the fixed-sized array.